packages/api/src/external/commonwell/document/shared.ts

```diff
@@ -39,3 +48,46 @@ export function getFileExtension(value: string | undefined): string {
   const fileExtension = extension(value);
   return fileExtension ? `.${fileExtension}` : "";
 }
+
+export function getExtraDocRefInfo(content: DocumentContent | undefined): DocumentReferenceExtras {
+  const extraInfo: DocumentReferenceExtras = {};
+  const contained = content?.contained;
+  const context = content?.context;
+
+  if (contained) {
+    contained.forEach(item => {
+      const containedName = item.name;
+      if (item.resourceType === "Organization") {
+        if (typeof containedName === "string") {
+          extraInfo.organization = {
+            name: containedName,
+          };
+        }
+      } else if (item.resourceType === "Practitioner" && containedName) {
+        if (Array.isArray(containedName)) {
+          const name = containedName[0];
+          if (name) addPractitionerName(extraInfo, name);
+        } else if (typeof containedName === "object") {
+          addPractitionerName(extraInfo, containedName);
```

Comment:
I recommend returning a value than modifying the `extrInfo` parameter, its a step towards immutability, which is good for SW maintainability. 🤓